tcp_check/datadog_checks/tcp_check/tcp_check.py

```diff
@@ -25,15 +27,26 @@ def __init__(self, name, init_config, instances):
         self.socket_type = None
         self._addr = None
 
+        ConfigurationErrorMsg = "Invalid `{}`; a {} must be specified."
+
+        self.ip_cache_last_ts = 0
+        self.ip_cache_duration = DEFAULT_IP_CACHE_DURATION
+        ip_cache_duration = instance.get('ip_cache_duration', None)
+        if ip_cache_duration is not None:
+            try:
+                self.ip_cache_duration = int(ip_cache_duration)
+            except Exception:
+                raise ConfigurationError(ConfigurationErrorMsg.format('ip_cache_duration', 'number'))
+
         port = instance.get('port', None)
         try:
             self.port = int(port)
         except Exception:
-            raise ConfigurationError("{} is not a correct port.".format(str(port)))
+            raise ConfigurationError(ConfigurationErrorMsg.format('port', 'number'))
         try:
             split_url = self.host.split(":")
         except Exception:  # Would be raised if url is not a string
-            raise ConfigurationError("A valid url must be specified")
+            raise ConfigurationError(ConfigurationErrorMsg.format('url', 'string'))
```

Comment:
```suggestion
            raise ConfigurationError(ConfigurationErrorMsg.format('url', 'string'))
```
Same as before, although in this case we were not printing the input value I think it's more informative if the message was something like:

`'{}' is an invalid `{}`; a {} must be specified.`